activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java

```diff
@@ -626,6 +626,102 @@ public void shouldGetProcessInstanceInitiatedByCurrentUser() {
         shouldNotGetProcessInstance(processInstanceInitiatedByUser1.getId());
     }
 
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsTaskAssignee() {
+        ProcessInstance firstProcessInstanceWithTestUserAssignedToATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.anAssignedTask("ATask", "testuser", firstProcessInstanceWithTestUserAssignedToATask);
+
+        ProcessInstance secondProcessInstanceWithTestUserAssignedToATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.anAssignedTask("ATask", "testuser", secondProcessInstanceWithTestUserAssignedToATask);
+
+        ProcessInstance thirdProcessInstanceWithRandomUserAssignedToTask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.anAssignedTask("ATask", "arandomuser", thirdProcessInstanceWithRandomUserAssignedToTask);
+
+        eventsAggregator.sendAll();
+
+        shouldGetProcessInstancesList(List.of(firstProcessInstanceWithTestUserAssignedToATask.getId(),
+            secondProcessInstanceWithTestUserAssignedToATask.getId()));
+    }
+
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsTaskCandidate() {
+        ProcessInstance firstProcessInstanceWithTestUserCandidateOfATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "testuser", firstProcessInstanceWithTestUserCandidateOfATask);
+
+        ProcessInstance secondProcessInstanceWithTestUserCandidateOfATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "testuser", secondProcessInstanceWithTestUserCandidateOfATask);
+
+        ProcessInstance thirdProcessInstanceWithRandomUserCandidateOfATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "arandomuser", thirdProcessInstanceWithRandomUserCandidateOfATask);
+
+        eventsAggregator.sendAll();
+
+        shouldGetProcessInstancesList(List.of(firstProcessInstanceWithTestUserCandidateOfATask.getId(),
+            secondProcessInstanceWithTestUserCandidateOfATask.getId()));
+    }
+
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsTaskAssigneeOrCandidate() {
```

Comment:
Aren't this 2 cases covered already in `shouldGetProcessInstancesWhenCurrentUserIsTaskAssignee` and `shouldGetProcessInstancesWhenCurrentUserIsTaskAssignee`?

Reply:
I wanted to make sure that query works using or.